src/olympia/files/tests/test_models.py

```diff
@@ -371,14 +371,16 @@ def test_get_review_status_display(self):
         assert file.get_review_status_display() == 'Approved'
 
     def test_file_status_signal(self):
-        addon = addon_factory(reviewer_flags={'auto_approval_disabled': True})
-        version = addon.current_version
+        user_factory(pk=settings.TASK_USER_ID)
+        addon = addon_factory(file_kw={'status': amo.STATUS_DISABLED})
+        version = addon.versions.get()
+        version.needshumanreview_set.create()
         assert not version.due_date
 
         version.file.update(status=amo.STATUS_AWAITING_REVIEW)
         assert version.reload().due_date
 
-        version.file.update(status=amo.STATUS_APPROVED)
+        version.file.update(status=amo.STATUS_DISABLED)
```

Comment:
Might be something obvious, but I'm missing why this test can to switch around the statuses leading to a due_date

Reply:
All this test does is prove we trigger a due date reset on file status change. Previously, because the add-on had `auto_approval_disabled`, the file started approved, so it didn't have a due date because auto-approval being disabled only had an impact on versions awaiting review. It was then changed to awaiting review, got a due date, and changed back to approved, so it lost it.

I've changed it to prove the reset was still happening on status change, but because now the only thing that matters to get a due date is having a NHR and not being disabled (or being signed), the file starts disabled, then is changed to awaiting review, then back to disabled.